classes/genimage.bbclass

```diff
@@ -92,19 +92,18 @@ do_genimage[depends] += "${@'${GENIMAGE_ROOTFS_IMAGE}:do_image_complete' if '${G
 GENIMAGE_TMPDIR  = "${WORKDIR}/genimage-tmp"
 GENIMAGE_ROOTDIR  = "${WORKDIR}/root"
 
-do_genimage[cleandirs] = "${GENIMAGE_TMPDIR} ${GENIMAGE_ROOTDIR}"
+do_genimage[cleandirs] = "${GENIMAGE_TMPDIR} ${GENIMAGE_ROOTDIR} ${B}"
 
 do_configure () {
     if grep -vq "@IMAGE@" ${WORKDIR}/genimage.config; then
         bbnote "genimage.config does not contain @IMAGE@ marker"
     fi
-
-    cp ${WORKDIR}/genimage.config ${B}/.config
 }
 
 do_genimage[dirs] = "${B}"
 
 fakeroot do_genimage () {
+    cp ${WORKDIR}/genimage.config ${B}/.config
```

Comment:
This looks much more straightforward than what we had before, yes. Having .config copied in do_genimage only will also automatically deprecate the following `.config.tmp` handling. There we then can simply use `.config`.